File: demo-service-impl/src/main/java/ru/zubmike/service/demo/logic/DictionaryLogic.java
package ru.zubmike.service.demo.logic;

import ru.zubmike.core.types.BasicDictItem;
import ru.zubmike.core.types.DictItem;
import ru.zubmike.service.demo.dao.PlanetarySystemDao;

import javax.inject.Inject;
import java.util.List;
import java.util.stream.Collectors;

public class DictionaryLogic {

	private final PlanetarySystemDao planetarySystemDao;

	@Inject
	public DictionaryLogic(PlanetarySystemDao planetarySystemDao) {
		this.planetarySystemDao = planetarySystemDao;
	}

	public List<BasicDictItem> getPlanetarySystems() {
		return planetarySystemDao.getAll()
				.stream()
				.map(DictionaryLogic::createBasicDictItem)
				.collect(Collectors.toList());
	}

	private static BasicDictItem createBasicDictItem(DictItem<Integer> item) {
		return new BasicDictItem(item.getId(), item.getName());
	}

}

File: demo-service-kotlin-impl/src/main/kotlin/ru/zubmike/service/demo/logic/DictionaryLogic.kt
package ru.zubmike.service.demo.logic

import ru.zubmike.core.types.BasicDictItem
import ru.zubmike.core.types.DictItem
import ru.zubmike.service.demo.dao.PlanetarySystemDao

class DictionaryLogic(
        private val planetarySystemDao: PlanetarySystemDao) {

    fun getPlanetarySystems(): List<BasicDictItem> {
        return planetarySystemDao.all
                .map { createBasicDictItem(it) }
    }

    private fun createBasicDictItem(item: DictItem<Int>): BasicDictItem {
        return BasicDictItem(item.id, item.name)
    }

}